spectral_cube/cube_utils.py

```diff
@@ -750,3 +751,105 @@ def bunit_converters(obj, unit, equivalencies=(), freq=None):
     else:
         # Slice along first axis to return a 1D array.
         return factors[0]
+
+def combine_headers(header1, header2):
+    '''
+    Given two Header objects, this function returns a fits Header of the optimal wcs. 
+
+    Parameters
+    ----------
+    header1 : astropy.io.fits.Header
+        A Header.
+    header2 : astropy.io.fits.Header
+        A Header.
+
+    Returns
+    -------
+    header : astropy.io.fits.Header
+        A header object of a field containing both initial headers. 
+        
+    '''
+    
+    from reproject.mosaicking import find_optimal_celestial_wcs
+    
+    head1 = header1.copy()
+    head1['NAXIS'] = 2
+    del head1['NAXIS3']
+    head2 = header2.copy()
+    head2['NAXIS'] = 2
+    del head2['NAXIS3']
+    
+    # Get wcs and shape of both headers
+    w1 = WCS(head1, naxis=2)
+    s1 = w1.array_shape
+    w2 = WCS(head2, naxis=2)
+    s2 = w2.array_shape
+    
+    # Get the optimal wcs and shape for both fields together
+    wcs_opt, shape_opt = find_optimal_celestial_wcs([(s1, w1), (s2, w2)], auto_rotate=False)
+    
+    # Make a new header using the optimal wcs and information from cubes
+    header = header1.copy()
+    header['NAXIS'] = 3
+    header['NAXIS1'] = shape_opt[1]
+    header['NAXIS2'] = shape_opt[0]
+    header['NAXIS3'] = header1['NAXIS3']
+    header.update(wcs_opt.to_header())
+    header['WCSAXES'] = 3
+    return header 
+
+def mosaic_cubes(cubes, spectral_block_size=100):
+    '''
+    This function reprojects cubes onto a common grid and combines them to a single field.  
+
+    Parameters
+    ----------
+    cubes : iterable
+        Iterable list of SpectralCube objects to reproject and add together. 
+    spectral_block_size : int
+        Block size so that reproject does not run out of memory. 
+
+    Outputs
+    -------
+    cube : SpectralCube
+        A spectral cube with the list of cubes mosaicked together.
+    '''
+    
+    cube1 = cubes[0]
+    cube1.allow_huge_operations = True
```

Comment:
Best not to do this.  It's up to the user to specify this before passing in the arguments.  Also, you've only done this for cube 1, but you'd need to for each.